chef_master/source/berkshelf.rst

```diff
@@ -3,7 +3,181 @@ About Berkshelf
+.. tag berkshelf_summary
+
+Berkshelf is a dependency manager for certain cookbook workflows that is included in the Chef development kit.
+
+.. note:: For new users, we strongly recommend using :doc:`Policyfiles </policyfile>` rather than Berkshelf.
+
+.. end_tag
+
+The Berksfile
+==============
+
+.. tag berksfile_summary
+
+A Berksfile describes the set of sources and dependencies needed to use a cookbook. It is used in conjunction with the ``berks`` command.
+
+.. end_tag
+
+Syntax
+-------
+A Berksfile is a ruby file, in which sources, dependencies, and options may be specified. Berksfiles are modelled closely on Bundler's Gemfile. The syntax is as follows:
+
+.. code-block:: ruby
+
+  source "https://supermarket.chef.io"
+  metadata
+  cookbook "NAME" [, "VERSION_CONSTRAINT"] [, SOURCE_OPTIONS]
+
+
+Metadata Keyword
++++++++++++++++++
+
+The ``metadata`` keyword causes Berkshelf to process the local cookbook metadata.
+This ensures that the dependencies of the cookbook are resolved by Berkshelf. Using the ``metadata`` keyword requires that the Berksfile be placed in the root of the cookbook, next to ``metadata.rb``.
+
+Source Keyword
++++++++++++++++
+
+A source defines where Berkshelf should look for cookbooks. Sources are processed in the order that they are defined in, and processing stops as soon as a suitable cookbook is found. A location can either be a :doc:`Supermarket <supermarket>` system, or a Chef Server.
+By default, a Berksfile has a source for Chef's public supermarket:
+
+.. code-block:: ruby
+
+  source "https://supermarket.chef.io"
+
+To add a private supermarket, which will be preferred:
+
+.. code-block:: ruby
+
+  source "https://supermarket.example.com"
+  source "https://supermarket.chef.io"
+
+To add a Chef Server:
+
+.. code-block:: ruby
+
+  source "https://supermarket.chef.io"
+  source :chef_server
+
+The location and authentication details for the Chef Server will be taken from the user's ``knife.rb``.
+
+Cookbook Keyword
+++++++++++++++++++
+
+The ``cookbook`` keyword allows the user to define where a cookbook is installed from, or to set additional version constraints. It can also be used to install additonal cookbooks, for example to use during testing.
+
+The format of a ``cookbook`` stanza is as follows:
+
+.. code-block
+
+  cookbook "NAME" [, "VERSION_CONSTRAINT"] [, SOURCE_OPTIONS]
+
+The simplest form is:
+
+.. code-block:: ruby
+
+  cookbook "library-cookbook"
+
+This ensures that a cookbook named ``my-cookbook`` is installed by berkshelf.
+
+Version constraints are the second parameter:
+
+.. code-block:: ruby
+
+  cookbook "library-cookbook", "~> 0.1.1"
+
+These are identical to the version constraints in a :ref:`cookbook metadata file <cookbook_version_constraints>`.
```

Comment:
We have a cookbook versions page that might better explain the constraints. It's also huge and probably a bit too complex